In [5]:
import pandas as pd
import matplotlib.pyplot as plt

## Map countries to remove duplication

Countries must be mapped as there are multiple representations of each country in the dataset.

In [6]:
country_mapping = {
    'CA': 'Canada',
    'AU': 'Australia',
    'CI': "Cote D'ivoire",
    'PK': 'Pakistan',
    'AF': 'Afghanistan',
    'KE': 'Kenya',
    'MQ': 'Martinique',
    'GB': 'United Kingdom',
    'SK': 'Slovakia',
    'CZ': 'Czech Republic',
    'DE': 'Germany',
    'FR': 'France',
    'HR': 'Croatia',
    'PL': 'Poland',
    'ZA': 'South Africa',
    'CH': 'Switzerland',
    'NG': 'Nigeria',
    'SE': 'Sweden',
    'IE': 'Ireland',
    'ES': 'Spain',
    'MX': 'Mexico',
    'IN': 'India',
    'LT': 'Lithuania',
    'GH': 'Ghana',
    'BD': 'Bangladesh',
    'SI': 'Slovenia',
    'HK': 'Hong Kong',
    'NO': 'Norway',
    'NL': 'Netherlands',
    "Cote D'ivoire": "Cote D'ivoire",
    'DO': 'Dominican Republic',
    'PT': 'Portugal',
    'PH': 'Philippines',
    'MY': 'Malaysia',
    'SG': 'Singapore',
    'AT': 'Austria',
    'IT': 'Italy',
    'DK': 'Denmark',
    'TR': 'Turkey',
    'BR': 'Brazil',
    'MA': 'Morocco',
    'TH': 'Thailand',
    'NZ': 'New Zealand',
    'JM': 'Jamaica',
    'KW': 'Kuwait',
    'VN': 'Vietnam',
    'QA': 'Qatar',
    'PE': 'Peru',
    'AL': 'Albania',
    'BB': 'Barbados',
    'RO': 'Romania',
    'BN': 'Brunei',
    'BH': 'Bahrain',
    'U.A.E.': 'United Arab Emirates',
    'RU': 'Russia',
    'CL': 'Chile',
    'MV': 'Maldives',
    'MT': 'Malta',
    'PG': 'Papua New Guinea',
    'UZ': 'Uzbekistan',
    'BG': 'Bulgaria',
    'CO': 'Colombia',
    'LR': 'Liberia',
    'AG': 'Antigua and Barbuda',
    'ZM': 'Zambia',
    'MU': 'Mauritius',
    'AI': 'Anguilla',
    'SA': 'Saudi Arabia',
    'TZ': 'Tanzania',
    'EE': 'Estonia',
    'KR': 'South Korea',
    'KH': 'Cambodia',
    'KY': 'Cayman Islands',
    'JO': 'Jordan',
    'BT': 'Bhutan',
    'MD': 'Moldova',
    'HT': 'Haiti',
    'VC': 'Saint Vincent and The Grenadine',
    'DM': 'Dominica',
    'ME': 'Montenegro',
    'MK': 'North Macedonia',
    'MZ': 'Mozambique',
    'HU': 'Hungary',
    'MN': 'Mongolia',
    'IS': 'Iceland',
    'AR': 'Argentina',
    'EC': 'Ecuador',
    'GU': 'Guam',
    'AW': 'Aruba',
    'PF': 'French Polynesia',
    'LS': 'Lesotho',
    'LU': 'Luxembourg',
    'KN': 'Saint Kitts and Nevis',
    'SX': 'Netherlands',
    'Sint Maarten (Dutch part)' : 'Netherlands',
    'VI': 'Virgin Islands (U.S.)',
    'PY': 'Paraguay',
    'FJ': 'Fiji',
    'GD': 'Grenada',
    'GY': 'Guyana',
    'MF': 'France',
    'TW': 'Taiwan',
    'MW': 'Malawi',
    'GF': 'French Guiana',
    'BJ': 'Benin',
    'GP': 'Guadeloupe',
    'TC': 'Turks and Caicos Islands',
    'BZ': 'Belize',
    'SV': 'El Salvador',
    'BO': 'Bolivia',
    'SN': 'Senegal',
    'GN': 'Guinea',
    'CR': 'Costa Rica',
    'ML': 'Mali',
    'BM': 'Bermuda',
    'BQ': 'Bonaire, Sint Eustatius and Saba',
    'AD': 'Andorra',
    'OM': 'Oman',
    'PM': 'Saint Pierre and Miquelon',
    'RE': 'Reunion',
    'LC': 'Saint Lucia',
    'NC': 'New Caledonia',
    'Kosovo': 'Kosovo',
    'Japan': 'Japan',
    'China': 'China',
    'Venezuela': 'Venezuela',
    'Trinidad and Tobago': 'Trinidad and Tobago',
    'United+Kingdom': 'United Kingdom',
    'Hong+Kong': 'Hong Kong',
    'South+Africa': 'South Africa',
    'Puerto Rico': 'Puerto Rico',
    'Sao Tome and Principe': 'Sao Tome and Principe',
    'French Polynesi': 'French Polynesia',
    'United Arab Emi': 'United Arab Emirates',
    'Hong Kong S.A.R.': 'Hong Kong',
    'Runion--trunc': 'Reunion',
    'Virgin Islands': 'Virgin Islands (U.S.)',
    'Iran': 'Iran',
    'Bosnia and Herz': 'Bosnia and Herzegovina',
    'Saint Vincent a': 'Saint Vincent and The Grenadine',
    'Papua New Guine': 'Papua New Guinea',
    'Myanmar': 'Myanmar',
    'Zimbabwe': 'Zimbabwe',
    'Sierra Leone': 'Sierra Leone',
    'Syria': 'Syria',
    'Trinidad and To': 'Trinidad and Tobago',
    'Palestine': 'Palestine',
    'Burundi': 'Burundi',
    'Laos': 'Laos',
    'Dominican Repub': 'Dominican Republic',
    'United States M': 'United States',
    'Antigua and Bar': 'Antigua and Barbuda',
    'Palau': 'Palau',
    'Djibouti': 'Djibouti',
    'Curaçao': 'Curacao',
    'Swaziland': 'Eswatini',
    'Palestinian Authority': 'Palestine',
    'Mauritania': 'Mauritania',
    'Monaco': 'Monaco',
    'Åland Islands': 'Aland Islands',
    'Gibraltar': 'Gibraltar',
    'Armenia': 'Armenia',
    'Seychelles': 'Seychelles',
    'Paraguay': 'Paraguay',
    'Equatorial Guin': 'Equatorial Guinea',
    'Kyrgyzstan': 'Kyrgyzstan',
    'Faroe Islands': 'Faroe Islands',
    'Antarctica': 'Antarctica',
    'United States': 'United States',
    'Wallis and Futu': 'Wallis and Futuna',
    'Eritrea': 'Eritrea',
    'Turks and Caico': 'Turks and Caicos Islands',
    'land Islands--trunc': 'land Islands',  # Truncated, assuming Åland Islands
    'Congo (Republic': 'Republic of the Congo',
    'Chad': 'Chad',
    'Samoa': 'Samoa',
    'Sao Tome and Pr': 'Sao Tome and Principe',
    'Tonga': 'Tonga',
    'Tahiti': 'French Polynesia',  # Assuming Tahiti is part of French Polynesia
    'Åland Islands': 'Aland Islands',
    'Nauru': 'Nauru',
    'Taiwan': 'Taiwan',
    'Solomon Islands': 'Solomon Islands',
    'Comoros': 'Comoros',
    'NigeriaHouse no 3 Pre': 'Nigeria',  # Unclear data, assuming Nigeria
    'Tajikistan': 'Tajikistan',
    'Nicaragua': 'Nicaragua',
    'Curaçao': 'Curacao',
    'Kiribati': 'Kiribati',
    'Saint Martin (F': 'Saint Martin',
    'Bonaire, Sint E': 'Bonaire, Sint Eustatius and Saba',
    'Saint Kitts and': 'Saint Kitts and Nevis',  # Incomplete entry, assuming Nevis
    'South Georgia a': 'South Georgia and The South Sandwich Islands',  # Incomplete entry, assuming South Sandwich Islands
    'British Indian': 'British Indian Ocean Territory',
    'Guernsey': 'Guernsey',
    'Liechtenstein': 'Liechtenstein',
    'Greenland': 'Greenland',
    'French Southern': 'France',
    'YT' : 'France',
    'Korea': 'South Korea',  # Assuming South Korea
    'BS': 'The Bahamas',
    'ID': 'Indonesia',
    'Réunion': 'France',  # Assuming Réunion is part of France
    'AE': 'United Arab Emirates'
}

In [94]:
# Read the CSV file into a DataFrame
login_dataframe = pd.read_csv("login.csv")

# Remap the 'country' column using the country_mapping dictionary
login_dataframe['country'] = login_dataframe['country'].map(country_mapping).fillna(login_dataframe['country'])
login_dataframe['reg_date'] = pd.to_datetime(login_dataframe['reg_date'], unit='s')

# Remove the 'account_currency' columns
login_dataframe.drop(columns=['account_currency'], inplace=True)

login_dataframe.head()

login  country            reg_date
0  457547  Romania 2021-02-25 00:15:32
1  474589   Canada 2021-01-07 02:44:02
2  504321   Canada 2020-10-14 02:31:50
3  504322   Canada 2020-10-15 04:35:45
4  504326   Canada 2020-10-19 07:39:12

In [86]:
login_dataframe.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 40512 entries, 0 to 40511
Data columns (total 3 columns):
 #   Column    Non-Null Count  Dtype         
---  ------    --------------  -----         
 0   login     40512 non-null  int64         
 1   country   40505 non-null  object        
 2   reg_date  40512 non-null  datetime64[ns]
dtypes: datetime64[ns](1), int64(1), object(1)
memory usage: 949.6+ KB


## Load trades dataframe and merge with login

Only accounts that have registered within the timeframe of trades.csv are considered. This is because we do not have complete trading data for accounts before that and do not want to misrepresent data which may negatively influence a predictive model.

In [95]:
# Load and prepare the trades data frame
trades_dataframe = pd.read_csv("trades.csv")
trades_dataframe['open_time'] = pd.to_datetime(trades_dataframe['open_time'], unit='s')
trades_dataframe['close_time'] = pd.to_datetime(trades_dataframe['close_time'], unit='s')


In [111]:
# Range of trades_dataframe
min_trade_date1 = trades_dataframe['open_time'].min()
max_trade_date1 = trades_dataframe['close_time'].max()
print(max_trade_date1 - min_trade_date1)

422 days 23:55:32


In [88]:
trades_dataframe.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4521777 entries, 0 to 4521776
Data columns (total 16 columns):
 #   Column       Dtype         
---  ------       -----         
 0   ticket       int64         
 1   login        int64         
 2   symbol       object        
 3   cmd          int64         
 4   volume       float64       
 5   open_time    datetime64[ns]
 6   open_price   float64       
 7   close_time   datetime64[ns]
 8   close_price  float64       
 9   tp           float64       
 10  sl           float64       
 11  reason       int64         
 12  commission   float64       
 13  swaps        float64       
 14  profit       float64       
 15  volume_usd   float64       
dtypes: datetime64[ns](2), float64(9), int64(4), object(1)
memory usage: 552.0+ MB


In [89]:
# Check no. unique accounts in trades
trades_dataframe['login'].nunique()

11976

In [96]:
# Identify the minimum and maximum trade dates
min_trade_date = trades_dataframe['open_time'].min()
max_trade_date = trades_dataframe['close_time'].max()

# Filter the login dataframe to include only accounts registered within the trade dates range
filtered_login_dataframe = login_dataframe[(login_dataframe['reg_date'] >= min_trade_date) & (login_dataframe['reg_date'] <= max_trade_date)]

# Merge the filtered login data with the trades data
merged_df = pd.merge(trades_dataframe, filtered_login_dataframe, on='login', how='inner')

## Feature engineering and merging

TODO: justify feature engineering

In [97]:
# Group by 'login' to get the count of tickets
total_tickets_per_login = merged_df.groupby('login')['ticket'].count()

# Calculate the number and percentage of buy tickets
buy_tickets_per_login = merged_df[merged_df['cmd'] == 0].groupby('login')['ticket'].size().astype('int64').fillna(0)
percentage_buys = (buy_tickets_per_login / total_tickets_per_login) * 100

# Calculate averages for various trade parameters
average_volume_per_login = merged_df.groupby('login')['volume'].mean()
average_volume_usd_per_login = merged_df.groupby('login')['volume_usd'].mean()
average_open_price_per_login = merged_df.groupby('login')['open_price'].mean()
average_close_price_per_login = merged_df.groupby('login')['close_price'].mean()
average_tp_per_login = merged_df.groupby('login')['tp'].mean()
average_sl_per_login = merged_df.groupby('login')['sl'].mean()
average_commission_per_login = merged_df.groupby('login')['commission'].mean()
average_swaps_per_login = merged_df.groupby('login')['swaps'].mean()
average_profit_per_login = merged_df.groupby('login')['profit'].mean()

# Find the most common reason per account
reason_per_login = merged_df.groupby('login')['reason'].apply(lambda x: x.value_counts().idxmax())

# Compile these metrics into a single dataframe
result_dataframe = pd.DataFrame({
    'Total_Trades': total_tickets_per_login,
    'Buy_Percentage': percentage_buys,
    'Average_Volume': average_volume_per_login,
    'Average_Volume_USD': average_volume_usd_per_login,
    'Average_Open_Price': average_open_price_per_login,
    'Average_Close_Price': average_close_price_per_login,
    'Average_TP': average_tp_per_login,
    'Average_SL': average_sl_per_login,
    'Most_Common_Trading_Method': reason_per_login,
    'Average_Commission': average_commission_per_login,
    'Average_Swaps': average_swaps_per_login,
    'Average_Profit': average_profit_per_login
})

In [98]:
final_dataset = pd.merge(login_dataframe, result_dataframe, on='login', how='inner')

In [100]:
final_dataset.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8167 entries, 0 to 8166
Data columns (total 15 columns):
 #   Column                      Non-Null Count  Dtype         
---  ------                      --------------  -----         
 0   login                       8167 non-null   int64         
 1   country                     8167 non-null   object        
 2   reg_date                    8167 non-null   datetime64[ns]
 3   Total_Trades                8167 non-null   int64         
 4   Buy_Percentage              7312 non-null   float64       
 5   Average_Volume              8167 non-null   float64       
 6   Average_Volume_USD          8167 non-null   float64       
 7   Average_Open_Price          8167 non-null   float64       
 8   Average_Close_Price         8167 non-null   float64       
 9   Average_TP                  8167 non-null   float64       
 10  Average_SL                  8167 non-null   float64       
 11  Most_Common_Trading_Method  8167 non-null   int64       

## Longevity calculations and merging

Longevity is defined as close time of last trade minus open time of first trade for a given account (definition confirmed by Carol). 

For longevity, it's also important to consider whether an account is actively trading or is inactive. An account is considered 'active' if it trades within the mosty recent month of the dataset, and inactive if it doesn't. Accounts that have registered in the most recent month of the dataset have been removed to avoid confusion.

We do not want to label an account that becomes inactive after 1 month the same way as an account that has actively traded for 1 month, but is still ongoing. We can't define longevity for these 'active' accounts as we do not know when they will stop trading. Therefore, it stands to reason that due to having a known duration, 'inactive' accounts are better indicators of longevity, at least in the short term.

Because of this, the dataset is binned into 5 categories, where only 'inactive' accounts are included up until 360 days (trades.csv has a range of 422 days). For '360+' both active and inactive are included as this bin is open ended.

In [101]:
# Calculate the first open time and last close time for each account
first_open_time_per_login = merged_df.groupby('login')['open_time'].min()
last_close_time_per_login = merged_df.groupby('login')['close_time'].max()

# Calculate longevity as the difference in days between the last close time and first open time
longevity_per_login = (last_close_time_per_login - first_open_time_per_login).dt.days

# Define the most recent month based on the latest trade in the dataset
most_recent_month_start = merged_df['close_time'].max().replace(day=1)

# Determine active accounts (trading in the most recent month)
active_accounts = merged_df[merged_df['close_time'] >= most_recent_month_start].groupby('login').size().index

# Mark accounts as active or inactive
longevity_per_login = longevity_per_login.to_frame(name='longevity')
longevity_per_login['active'] = longevity_per_login.index.isin(active_accounts)

# Exclude accounts that were registered in the most recent month
valid_accounts = login_dataframe[login_dataframe['reg_date'] < most_recent_month_start]

# Bin the longevity values
bins = [-1, 30, 90, 180, 270, 360, float('inf')]
longevity_per_login['longevity_bin'] = pd.cut(longevity_per_login['longevity'], bins=bins, labels=False)

# Exclude active accounts from all bins except for 360+
longevity_per_login = longevity_per_login[(longevity_per_login['active'] == False) | (longevity_per_login['longevity_bin'] == 5)]

In [105]:
# Merge longevity information with result_dataframe
result_dataframe = final_dataset.reset_index().merge(longevity_per_login, on='login', how='inner')

In [107]:
result_dataframe.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5725 entries, 0 to 5724
Data columns (total 19 columns):
 #   Column                      Non-Null Count  Dtype         
---  ------                      --------------  -----         
 0   index                       5725 non-null   int64         
 1   login                       5725 non-null   int64         
 2   country                     5725 non-null   object        
 3   reg_date                    5725 non-null   datetime64[ns]
 4   Total_Trades                5725 non-null   int64         
 5   Buy_Percentage              4914 non-null   float64       
 6   Average_Volume              5725 non-null   float64       
 7   Average_Volume_USD          5725 non-null   float64       
 8   Average_Open_Price          5725 non-null   float64       
 9   Average_Close_Price         5725 non-null   float64       
 10  Average_TP                  5725 non-null   float64       
 11  Average_SL                  5725 non-null   float64     

Filtering accounts in the ways described above has reduced our unique accounts from 8167 to 5724

In [108]:
result_dataframe.head()

index   login    country            reg_date  Total_Trades  Buy_Percentage  \
0      1  524978    Austria 2023-06-07 05:58:36          1392       47.485632   
1      2  524979     France 2023-06-07 06:17:30          2194       49.635369   
2      4  760487  Singapore 2023-01-04 08:48:24            69       15.942029   
3      5  804664   Malaysia 2023-09-24 07:02:49            85       63.529412   
4      6  804687  Australia 2023-01-10 10:26:00           484       51.239669   

   Average_Volume  Average_Volume_USD  Average_Open_Price  \
0        1.230632        4.044965e+08          513.912144   
1        0.013943        6.724644e+03        20467.746053   
2        0.012609        2.783882e+03        11010.476377   
3        0.018706        5.490400e+03         1265.465529   
4        0.019773        5.271755e+05          433.944222   

   Average_Close_Price    Average_TP    Average_SL  \
0           513.898092    297.756577    168.537030   
1         20466.447288   2818.566659  12916.827370   
2         11077.594638  10337.456522    168.292464   
3          1266.228878      0.000000      0.000000   
4           431.902211    323.711256    431.677983   

   Most_Common_Trading_Method  Average_Commission  Average_Swaps  \
0                           1           -6.429425      -1.366882   
1                           5            0.000000      -0.111285   
2                           1            0.000000       0.002899   
3                           5            0.000000      -0.023176   
4                           5           -0.151550      -0.015021   

   Average_Profit  longevity  active  longevity_bin  
0       23.616724        142   False              2  
1       -0.837867        107   False              2  
2      -13.648986         13   False              0  
3       -0.524588          9   False              0  
4       -0.319339        372   False              5

In [117]:
result_dataframe['longevity_bin'].value_counts()

longevity_bin
0    3575
1    1216
2     595
3     214
5      69
4      56
Name: count, dtype: int64

Distribution of longevity. Be aware that 5 appears before 4, because there are more accounts in that bin (most likely due to including both active and inactive accounts)

In [119]:
# Filter rows where 'longevity_bin' is 5
filtered_df = result_dataframe[result_dataframe['longevity_bin'] == 5]

# Count the occurrences of True and False in the 'active' column for the filtered rows
active_counts = filtered_df['active'].value_counts()

# Display the counts
print(active_counts)

active
True     63
False     6
Name: count, dtype: int64


Distribution of '360+' days by activity status.

## Final pre-processing

Simple pre-processing including converting less important trading methods to 'Other' and then reverse encoding, filling NaN values, and reording columns.

TODO: final pre-processing

In [120]:
(result_dataframe['Total_Trades'] == 0).sum()

0

In [121]:
result_dataframe['Most_Common_Trading_Method'].value_counts()

Most_Common_Trading_Method
5    2527
0    1957
1    1088
6     142
3      10
4       1
Name: count, dtype: int64

In [126]:
# Filter out the dataset
result_dataframe['Most_Common_Trading_Method'] = result_dataframe['Most_Common_Trading_Method'].apply(lambda x: x if x in [0, 1, 5] else 7)

# Rename the codes to strings
result_dataframe['Trading_Method'] = result_dataframe['Most_Common_Trading_Method'].map({0: 'Client', 1: 'Expert', 5: 'Mobile', 7:'Other'})

# Fill NaN values with appropriate strategy (mean, median, etc.)
result_dataframe.fillna(0, inplace=True)

In [128]:
# Get a list of all column names except 'longevity' and 'longevity_bin'
cols = [col for col in result_dataframe.columns if col not in ['longevity', 'longevity_bin']]

# Append 'longevity' and 'longevity_bin' to the end of the column list
cols += ['longevity', 'longevity_bin']

# Reorder the DataFrame based on the new column list
final_dataset = result_dataframe[cols]

In [132]:
final_dataset.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5725 entries, 0 to 5724
Data columns (total 20 columns):
 #   Column                      Non-Null Count  Dtype         
---  ------                      --------------  -----         
 0   index                       5725 non-null   int64         
 1   login                       5725 non-null   int64         
 2   country                     5725 non-null   object        
 3   reg_date                    5725 non-null   datetime64[ns]
 4   Total_Trades                5725 non-null   int64         
 5   Buy_Percentage              5725 non-null   float64       
 6   Average_Volume              5725 non-null   float64       
 7   Average_Volume_USD          5725 non-null   float64       
 8   Average_Open_Price          5725 non-null   float64       
 9   Average_Close_Price         5725 non-null   float64       
 10  Average_TP                  5725 non-null   float64       
 11  Average_SL                  5725 non-null   float64     

In [129]:
final_dataset.head()

index   login    country            reg_date  Total_Trades  Buy_Percentage  \
0      1  524978    Austria 2023-06-07 05:58:36          1392       47.485632   
1      2  524979     France 2023-06-07 06:17:30          2194       49.635369   
2      4  760487  Singapore 2023-01-04 08:48:24            69       15.942029   
3      5  804664   Malaysia 2023-09-24 07:02:49            85       63.529412   
4      6  804687  Australia 2023-01-10 10:26:00           484       51.239669   

   Average_Volume  Average_Volume_USD  Average_Open_Price  \
0        1.230632        4.044965e+08          513.912144   
1        0.013943        6.724644e+03        20467.746053   
2        0.012609        2.783882e+03        11010.476377   
3        0.018706        5.490400e+03         1265.465529   
4        0.019773        5.271755e+05          433.944222   

   Average_Close_Price    Average_TP    Average_SL  \
0           513.898092    297.756577    168.537030   
1         20466.447288   2818.566659  12916.827370   
2         11077.594638  10337.456522    168.292464   
3          1266.228878      0.000000      0.000000   
4           431.902211    323.711256    431.677983   

   Most_Common_Trading_Method  Average_Commission  Average_Swaps  \
0                           1           -6.429425      -1.366882   
1                           5            0.000000      -0.111285   
2                           1            0.000000       0.002899   
3                           5            0.000000      -0.023176   
4                           5           -0.151550      -0.015021   

   Average_Profit  active Trading_Method  longevity  longevity_bin  
0       23.616724   False         Expert        142              2  
1       -0.837867   False         Mobile        107              2  
2      -13.648986   False         Expert         13              0  
3       -0.524588   False         Mobile          9              0  
4       -0.319339   False         Mobile        372              5

In [130]:
final_dataset.tail()

index       login       country            reg_date  Total_Trades  \
5720   8157    88944953        Canada 2023-01-12 02:52:40            54   
5721   8159    88944971      Malaysia 2023-02-08 08:52:48           442   
5722   8161    88945034     Singapore 2023-04-12 13:01:26           479   
5723   8163    88945038     Singapore 2023-05-05 11:25:20          1306   
5724   8165  1000054838  Saudi Arabia 2023-09-26 17:11:57          4125   

      Buy_Percentage  Average_Volume  Average_Volume_USD  Average_Open_Price  \
5720       37.037037        0.041667        1.621176e+07          648.476864   
5721       46.153846        1.389367        1.249156e+09           23.466324   
5722       40.083507        0.043278        1.522641e+07         1549.084244   
5723       58.805513        0.078913        1.824879e+04            1.225047   
5724       56.533333        0.105981        4.312099e+04         2302.209052   

      Average_Close_Price  Average_TP   Average_SL  \
5720           648.428424    1.797778     0.020074   
5721            23.470589    0.402277     4.315796   
5722          1548.833586  842.501651   496.148427   
5723             1.225408    1.193976     0.243870   
5724          2302.325806   87.565501  1426.166116   

      Most_Common_Trading_Method  Average_Commission  Average_Swaps  \
5720                           5            0.000000      -0.007963   
5721                           1            0.000000      -5.085113   
5722                           1           -0.256347      -0.000271   
5723                           1           -0.946953      -0.169877   
5724                           1            0.000000      -0.026504   

      Average_Profit  active Trading_Method  longevity  longevity_bin  
5720       -0.175556    True         Mobile        381              5  
5721      -51.760588   False         Expert        100              2  
5722       -1.519937   False         Expert         46              1  
5723       -0.147504   False         Expert        253              3  
5724       -2.182613   False         Expert         85              1

## Export

In [131]:
final_dataset.to_csv('output_dataset.csv', index=False)